tests/contract/test_registry_management.py

```diff
@@ -0,0 +1,325 @@
+from eth_tester.backends.pyevm.main import (
+    get_default_account_keys,
+)
+
+from tests.handler.fixtures import (  # noqa: F401
+    smc_handler,
+)
+from handler.utils.web3_utils import (
+    mine,
+)
+
+
+def generate_notary_credential(notary_index):
```

Comment:
I think it could be cleaner with a simple class:
 ```python
class NotaryAccount:
    index = None
    def __init__(self, index):
        self.index = index

    @property
    def private_key:
        ...
    
    @property
    def canonical_address:
        ...

    @property
    def checksum_address:
        ...

    @classmethod
    def from_notary_index(cls, index):
        return cls(index)
```

Then in the test:
```python
notary_1 = NotaryAccount.from_index(1)
```